packages/trader/src/AppV2/Containers/Trade/trade.tsx

```diff
@@ -1,13 +1,191 @@
+            const other_category = updated_others.find(cat => cat.id === 'other');
+
+            if (other_category) {
+                other_category.items.unshift(item);
+            } else {
+                updated_others.push({
+                    id: 'other',
+                    items: [item],
+                });
+            }
+
+            updated_others.map(category => ({
+                ...category,
+                items: category.items.sort((a, b) => a.title.localeCompare(b.title)),
+            }));
+
+            return updated_others;
+        });
+    };
+
+    React.useEffect(() => {
+        const formatted_items = createArrayFromCategories(contract_types_list);
+
+        setOtherTradeTypes([
+            {
+                id: 'other',
+                items: formatted_items
+                    .filter(item => !pinned_trade_types[0]?.items.some(pinned_item => pinned_item.id === item.id))
+                    .sort((a, b) => a.title.localeCompare(b.title)),
+            },
+        ]);
+    }, [contract_types_list]);
+
+    React.useEffect(() => {
+        const saved_other_trade_types = localStorage.getItem('other_trade_types');
+        const saved_pinned_trade_types = localStorage.getItem('pinned_trade_types');
+
+        prepareTradeStore();
+
+        if (saved_other_trade_types) {
+            setOtherTradeTypes(JSON.parse(saved_other_trade_types));
+        }
+
+        if (saved_pinned_trade_types) {
+            setPinnedTradeTypes(JSON.parse(saved_pinned_trade_types));
+        }
+    }, []);
+
+    const savePinnedToLocalStorage = () => {
+        localStorage.setItem('pinned_trade_types', JSON.stringify(pinned_trade_types));
+        localStorage.setItem('other_trade_types', JSON.stringify(other_trade_types));
+        setOpen(false);
+    };
+
+    const handleOnDrag = (categories: ResultItem[]) => {
+        setPinnedTradeTypes(categories);
+    };
 
-const Trade = () => {
     return (
-        <BottomNav>
-            <Text size='sm'>Trade</Text>
+        <BottomNav className='trade-page'>
+            <ActionSheet.Root isOpen={open} onOpen={handleOpenTradeTypes} expandable={false}>
+                <ActionSheet.Trigger label='Edit Trade Types' />
+                <ActionSheet.Portal>
+                    <ActionSheet.Header title={<Localize i18n_default_text='Trade types' />} />
+                    <ActionSheet.Content className='mock-action-sheet--content'>
+                        <DraggableList
+                            categories={pinned_trade_types}
+                            onRightIconClick={handleRemovePinnedClick}
+                            onSave={savePinnedToLocalStorage}
+                            onDrag={handleOnDrag}
+                        />
+                        <TradeTypeList categories={other_trade_types} onRightIconClick={handleAddPinnedClick} />
+                    </ActionSheet.Content>
+                </ActionSheet.Portal>
+            </ActionSheet.Root>
```

Comment:
We should extract this out to a component and add test. There will be a lot of other components on trade page

Reply:
Changes in this PR for trade file are more or less for testing purpose I would say

Reply:
As mentioned above, I will work on moving it out in the next PR since this component will be behind another component